### PR TITLE
feat: create list feature

### PR DESCRIPTION
**Pull request by OgDev-01**
## Description
This pull request includes various features, fixes, and refactors to enhance the functionality and user experience of the Insights hub. It introduces the "hub list" feature, adds fetch hooks for lists and contributors, implements a filter chip component, and creates a contributor hub header component. Additionally, it includes several bug fixes, style updates, and code cleanups. Overall, these changes aim to improve the performance, layout, and navigation of the Insights hub.

### Things to take note of

- The range filter is not functional yet but will be added in my next PR
- limit filter works 
- the other advanced filters will be coming up in follow up PRs

### coming up next
- https://github.com/open-sauced/app/issues/1672
- https://github.com/open-sauced/app/issues/1671

## What type of PR is this? (check all applicable)

- [ ] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI
- [ ] 📦 Chore (Release)
- [ ] ⏩ Revert

## Related Tickets & Documents

related to #1665 #1663 

### Components
image

https://github.com/open-sauced/app/assets/62995161/498b61a3-84b1-4c7b-b3f9-681daf1cf196

## Added tests?

- [ ] 👍 yes
- [x] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help

## Added to documentation?

- [ ] 📜 README.md
- [ ] 📓 docs.opensauced.pizza
- [ ] 🍕 dev.to/opensauced
- [x] 📕 storybook
- [ ] 🙅 no documentation needed

### CODE REVIEW

**Comment by OgDev-01** [comment]
Side note:

I pulled changes from https://github.com/open-sauced/app/pull/1682 to expedite implementation while it undergoes review.

**Comment by OgDev-01** [comment]
Apologizing for adding too many changes on this pull request 🙏 🤗. Moving forward, I'll try not to make too many changes in 1 PR 👍

File: components/atoms/FilterChip/filter-chip.tsx [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+interface FilterChipProps {
+  className?: string;
+  items: string[];
+  onClear: () => void;
+}
+const FilterChip = ({ className, items, onClear }: FilterChipProps) => {
+  return (
+    <div
+      className={clsx(
+        "p-2.5 rounded-xl bg-sauced-light w-max text-sauced-orange max-w-[200px] flex items-center",
```
Comment (NsdHSO) [inline components/atoms/FilterChip/filter-chip.tsx:14]: `w-52` -> `208px`

File: components/molecules/ContributorListTableHeader/contributor-list-table-header.tsx [hunk 2]
```diff
@@ -2,8 +2,15 @@ import clsx from "clsx";
 import React from "react";
 import TableTitle from "components/atoms/TableTitle/table-title";
 import { classNames } from "components/organisms/RepositoriesTable/repositories-table";
+import Checkbox from "components/atoms/Checkbox/checkbox";
 
-const ContributorListTableHeaders = () => {
+interface ContributorListTableHeadersProps {
+  selected?: boolean;
+
```
Comment (NsdHSO) [inline components/molecules/ContributorListTableHeader/contributor-list-table-header.tsx:9]:
```suggestion
```

File: components/molecules/ContributorListTableHeader/contributor-list-table-header.tsx [hunk 3]
```diff
@@ -17,6 +24,15 @@ const ContributorListTableHeaders = () => {
       </div>
       {/* Desktop */}
       <div className="hidden gap-6 px-6 py-4 border rounded-t-lg md:flex bg-light-slate-3">
+        {handleOnSelectAllContributor && (
+          <Checkbox
+            checked={selected ? true : false}
+            // disabled={!user}
```
Comment (NsdHSO) [inline components/molecules/ContributorListTableHeader/contributor-list-table-header.tsx:30]: why do we need these lines of code ???

File: components/molecules/HubContributorsHeader/hub-contributors-header.tsx [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+              onTitleChange?.(title);
+            }}
+          />
+          <Text className="text-light-slate-9">Select contributors to add to your list</Text>
+        </div>
+        <div className="flex flex-row items-center justify-center gap-6 header-info ">
+          <div className="flex items-center gap-2">
+            <span
+              className={clsx(
+                "w-max min-w-[1.4rem] h-6  p-3 text-sm items-center flex place-content-center rounded-full",
```
Comment (NsdHSO) [inline components/molecules/HubContributorsHeader/hub-contributors-header.tsx:52]: `w-6` -> `1.5rem`

File: components/molecules/HubContributorsHeader/hub-contributors-header.tsx [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+            )}
+            variant="text"
+            onClick={onAddToList}
+          >
+            Add to List <FaPlus className="ml-2 text-lg" />
+          </Button>
+        </div>
+      </div>
+      <div className="flex justify-between w-full">
+        <div className="flex items-center gap-4"></div>
```
Comment (NsdHSO) [inline components/molecules/HubContributorsHeader/hub-contributors-header.tsx:75]: why do we need this empty element?

File: components/molecules/HubContributorsHeader/hub-contributors-header.tsx [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+            placeholder="10 per page"
+            options={[
+              { name: "10 per page", value: "10" },
+              { name: "20 per page", value: "20" },
+              { name: "30 per page", value: "30" },
+              { name: "40 per page", value: "40" },
+              { name: "50 per page", value: "50" },
+            ]}
+            className="!w-36 overflow-x-hidden"
+            onChange={function (limit: string): void {
```
Comment (NsdHSO) [inline components/molecules/HubContributorsHeader/hub-contributors-header.tsx:91]: why not arrow fct?

File: layouts/hub-contributors.tsx [hunk 7]
```diff
@@ -0,0 +1,69 @@
+import React from "react";
+import { useRouter } from "next/router";
+
+// import SkeletonWrapper from "components/atoms/SkeletonLoader/skeleton-wrapper";
+import Footer from "components/organisms/Footer/footer";
+// import Header from "components/organisms/Header/header";
+import TopNav from "components/organisms/TopNav/top-nav";
+// import HubContributorsHeader from "components/molecules/HubContributorsHeader/hub-contributors-header";
```
Comment (NsdHSO) [inline layouts/hub-contributors.tsx:8]:
```suggestion
```

File: pages/hub/lists/new.tsx [hunk 8]
```diff
@@ -1,7 +1,262 @@
...
+      await navigator.clipboard.writeText(url);
+      toast({ description: "Copied to clipboard", variant: "success" });
+    } catch (error) {
+      console.log(error);
+    }
+  };
+
+  return (
+    <Dialog open={isOpen}>
+      <div className="info-container container w-full min-h-[100px]">
```
Comment (NsdHSO) [inline pages/hub/lists/new.tsx:134]: `w-24` -> `96px`

**Comment by bdougie** [comment]
I suggest having the pencil removed when editing the title. 

Screen Shot 2023-09-16 at 7 57 46 AM

File: components/atoms/FilterChip/filter-chip.tsx (hunk 1 above)
Comment (OgDev-01) [reply components/atoms/FilterChip/filter-chip.tsx:14]: I'm using the `w-max` to resize the chip relative to its content and setting a limit to its growth with the `max-w-[200px]`. changing to `w-52` will alter the auto-resize of the chip. What I can do here is convert the `px` value to `rem`

File: components/molecules/HubContributorsHeader/hub-contributors-header.tsx (hunk 5 above)
Comment (OgDev-01) [reply components/molecules/HubContributorsHeader/hub-contributors-header.tsx:75]: More filters will be added there in coming PRs

**Review by jpmcb — COMMENTED** [review]
Got this very strange behavior on FireFox (but not chrome) where my list title ... seems to overlap? I'm not sure what's going on, hard to tell:

Screenshot 2023-09-18 at 3 24 34 PM

---

Also noticed two other things:
- When text is entered into the search bar, the "X" button causes the bar to "jump". Preferably, this would be smooth. For example, on Twitter when searching, the "X" button just appears within the search bar on the right without resizing the search bar itself
- When the dropdown menu is opened, it causes the scroll bar to disappear which shifts the page to shift over to the right (maybe related to https://github.com/open-sauced/app/issues/1696 ?)

https://github.com/open-sauced/app/assets/23109390/a0fafbc0-4ddd-42dd-92fb-634a4f2ddc1b

**Comment by OgDev-01** [comment]
> When text is entered into the search bar, the "X" button causes the bar to "jump". Preferably, this would be smooth. For example, on Twitter when searching, the "X" button just appears within the search bar on the right without resizing the search bar itself

Nice catch 👍 . However, fixing the search close icon is out of the scope of this PR. I'll hide the search for now since search is not implemented from the API atm. Opened a ticket for that here https://github.com/open-sauced/app/issues/1717

**Comment by jpmcb** [comment]
Sounds good! The only thing from me then is the overlapping titles in Firefox!

I noticed that it's only happening with longer list names:
Screenshot 2023-09-19 at 2 24 22 PM

But not short ones:
Screenshot 2023-09-19 at 2 23 24 PM

**Comment by OgDev-01** [comment]
> Sounds good! The only thing from me then is the overlapping titles in Firefox!
> 
> I noticed that it's only happening with longer list names: Screenshot 2023-09-19 at 2 24 22 PM
> 
> But not short ones: Screenshot 2023-09-19 at 2 23 24 PM

This has been fixed in the most recent updates.


>When the dropdown menu is opened, it causes the scroll bar to disappear which shifts the page over to the right (maybe related to https://github.com/open-sauced/app/issues/1696 ?)


this fix is still in progress for Select... hopefully it is merged soon https://github.com/radix-ui/primitives/pull/2253. We can move forward with this until the fix is ready to imlement

**Comment by brandonroberts** [comment]
@OgDev-01 Let's go ahead and get the page landed so we can reuse the contributors table in #1710. We can add filtering separately.

**Comment by OgDev-01** [comment]
Aside adding filters, This is ready for review cc: @open-sauced/engineering

**Review by brandonroberts — APPROVED** [review]
Needs more work on mobile, but its functional

**Comment by OgDev-01** [comment]
Will it be out of scope if I add a `Delete` flow for the list?

https://github.com/open-sauced/app/assets/62995161/76b049ed-d2b0-4676-a427-c0cf8826c412

Similar to what was done with insights pages

**Review by jpmcb — APPROVED** [review]
Overall looks great! 👏🏼 

---

I love the idea of a delete user flow. Typing `DELETE` works 👍🏼 I've also seen where you have to type the name of the resource your deleting to confirm it. I.e., if my list is named "My list", i'd have to type literally "My list" into the prompt to confirm the delete.

But I think if we need to get this PR in, we should look at that as a followup feature add.

**Pull request merged**